### ISSUE TITLE
Specifications for the omip2*/JRA55-do experiments need amendment

### ISSUE THREAD

**Issue by HiroyukiTsujino** [WCRP-CMIP/CMIP6_CVs#704]
Experience from multiple modeling centers with JRA55-do forcing dataset indicates that results are more robust if the model simulations are run for 6 cycles of forcing. For omip2 (and possibly for omip1), we are proposing to modify the protocol to run the model simulations for 6 cycles with the updated version (1958-2018).

**Comment by durack1** [comment]
@HiroyukiTsujino thanks for opening this issue.

I note that we have the following details for:

**experiment_id = omip2**
```
min_number_years_per_sim = 295 (1958-2017, 59yrs, by 5 cycles = 295)
```
```
description = Global ocean - sea-ice coupled experiment forced with the JRA55-do
inter-annually varying atmospheric and river data sets for years 1958-2017. Initial ocean
tracer fields are based on observations. All Priority=1 OMIP diagnostics are requested
for all five cycles of the 59-year forcing to quantify drift. All OMIP diagnostics
(Priority=1,2,3) are requested for the 5th cycle
```

**experiment_id = omip2-spunup**
```
min_number_years_per_sim = 295
```
```
description = Same as the omip2 experiment except that it is not initialized with observed
climatologies; rather it is initialized with results from at least a 2000-year spin up of the
coupled physical-biogeochemical models. The spin up simulations may be made with the
classic online or offline approach, or with tracer-acceleration techniques or fast solvers.
If an online approach is used, at the end of the 5th cycle of the JRA55-do forcing, the
model's physical fields should be reinitialized to the values at the start of the 3rd cycle
in order to avoid long-term drift in those fields and to assure that they will not diverge
greatly from physical fields in the omip2 simulation. The spin up also includes
radiocarbon to evaluate deep-ocean circulation
```

For this to be changed, we'll need the replacement text/entries for each field, in addition to any other field amendments, see [CMIP6_experiment_id.html](https://rawgit.com/WCRP-CMIP/CMIP6_CVs/master/src/CMIP6_experiment_id.html) or the direct json file entries [here](https://github.com/WCRP-CMIP/CMIP6_CVs/blob/master/CMIP6_experiment_id.json#L5727-L5832)

ping @stephengriffies @taylor13 @jamesorr

**Comment by StephenGriffies** [comment]
The 5-cycle recommendation was made during the March 2019 CLIVAR OMDP panel meeting in Florida.  Many groups at the meeting found it to be a suitable request.  However, there was not 100% agreement.  

The broader question concerns what is an optimal number of cycles.  I think that if most models reach a stationary ACC transport and AMOC after two cycles, then the recommendation would be for just two (or three) cycles.  For some reason, some of the new models run with JRA55-do appear to need six cycles, thus prompting the recommendation from @HiroyukiTsujino.

However, I doubt that six will be necessary or sufficient for all models.  Some may need less and some more.  

Perhaps we should have originally offered a drift criteria such as done for some OBGC protocols.  This point needs to be revisited in the future to allow for low-drift models to contribute with, say, only three cycles whereas high-drift models to participate with 10, for example.  Assessing the drift is important, and I suggest that we need no less than three to be confident a model is indeed low-drift.  

So after that rambling preamble, do we need to change the protocol?  I am unsure.  My general  suggestion is that we let groups self-assess their drift.  If they deem their models to be spun-up, then they do not need to run six cycles.  In contrast, for high-drift models they may need to run more cycles.  

The integrity of the comparison is not compromised if groups run different number of cycles.  Instead, what is important is that they run long enough to remove climate drift for those features that are studied.      

I hope to hear from others before moving forward or not on this recommendation from @HiroyukiTsujino.

**Comment by durack1** [comment]
@StephenGriffies in previous CORE-II analysis (e.g. [Griffies et al., 2009](https://doi.org/10.1016/j.ocemod.2014.03.004); [Danabasoglu et al., 2014](https://doi.org/10.1016/j.ocemod.2013.10.005)) only the last ~20 years (different for both studies) were used for the bulk of the analysis. In the current specification **min_number_years_per_simulation** of 295 (jra55-do/omip2) and 310 (core-II/omip1) years are requested, and this will need to be 300 for JRA55-do (1958 to 2018, with partial 2019 year data). I am wondering if this should be relaxed, as you note that some low-drift models will have a **stable** ocean state within as little as two cycles - so 120 years (1958-2018 x 2).

For ref - search for omip in [CMIP6_experiment_id.html](https://rawgit.com/WCRP-CMIP/CMIP6_CVs/master/src/CMIP6_experiment_id.html)

**Comment by StephenGriffies** [comment]
I agree, most of the CORE-II papers used the final 20 years.  That might be sufficient for OMIP as well, at least for some purposes.  But it would be nice to have at least one full cycle, the final cycle, from the models.  Having said that, only one cycle will not allow us to assess drift.  Also, JRA55-do is a better dataset and it might offer more interesting analysis for one full cycle, beyond the 20 years of CORE.  In particular, there are early suggestions that JRA55-do simulations provide a clear SST global warming signal, and that would be tough to study just with 20 years and just one cycle.  

So my recommendation is still to have full the final three cycles submitted with enough data to assess drift (temp, saln full depth annual means). All data for all cycles is generally not necessary.  Also, much of the newer 3d FAFMIP fields were only asked for as annuals and for the final cycle, I believe.  

I take it you are concerned with too much archive space and want to whittle down a bit?

**Comment by HiroyukiTsujino** [comment]
I concur with the recommendation from @StephenGriffies. This would be especially a good news for a high resolution model with small drift. 
I would just point out a small issue: the updated JRA55-do covers 1958-2018 and 1 cycle includes **61** years. So the minimum number of years would be 183 for a 3-cycle protocol and 366 for a 6-cycle protocol.

**Comment by StephenGriffies** [comment]
Thanks for reminding me that we have 61 years now.  That is a great advantage of JRA55-do, as @HiroyukiTsujino keeps giving us more data as time moves on!

**Comment by durack1** [comment]
@StephenGriffies if we're targeting drift specifically, remember we do have our `Odec` tables, see [cmip6-cmor-tables](https://github.com/PCMDI/cmip6-cmor-tables/blob/master/Tables/CMIP6_Odec.json) or the data request [Odec](http://clipc-services.ceda.ac.uk/dreq/u/MIPtable::Odec.html), in addition to the `Oyr` tables, so we could request the first (spinup) portion of the simulations at a reduced temporal resolution, and then the final full **61** year (1958 through 2018 for JRA55-do) at full temporal resolution `Omon` output?

**Comment by durack1** [comment]
@martinjuckes it would be great if we can get the proposed OMIP changes (which are not yet finalized) into [data request 01.00.31beta](https://earthsystemcog.org/projects/wip/drq_release_01_00_31) before this is released - what timeline are you expecting for this (01.00.31 release)?

@taylor13 ping

**Comment by StephenGriffies** [comment]
@durack1 the Odec is not fine enough in time to see the drift of interest; e.g., AMOC and ACC transports.  We instead need annuals. so Oyr is preferred.

**Comment by durack1** [comment]
@StephenGriffies ok so if this is the case, then the request would be for the final cycle (1958-2018, 61 years) full `Omon` output, and prior to this for the spinup phases (which may be 2 or 5 depending on the model configuration) we request the same output from `Oyr` - is this suitable? I note that the `Omon` table includes 292 variables, whereas the `Oyr` includes 122 - so defining the subset will be key.

I also wonder whether the `ocnBgchem` fields requested should also follow this protocol, or whether resolving the seasonal cycle is required - @jamesorr did you want to answer this for the `*-spunup` OMIP experiment variants?

**Comment by StephenGriffies** [comment]
@durack1 I concur.  Omon for final cycle and Oyr for prior (2 to 5) cycles.

**Comment by durack1** [comment]
@StephenGriffies @HiroyukiTsujino I note Hiroyuki has already amended his fork of the repo - are we happy with these amendments? If so, I'll pull these back into the master

**Comment by StephenGriffies** [comment]
@durack1 there has been lots of push-back from the OMDP concerning the number of cycles.  Not sure if you have been part of the email threads.   Many people (led by NCAR) are reticent to relax the required 5-cycles given the inertia in the system, and the interactions with the OBGC protocol.  I do not feel comfortably making this decision without full buy-in from the broader community.

**Comment by durack1** [comment]
@StephenGriffies I haven't been on these threads, so thanks for the heads up. I agree that without full support no changes should be made, not sure how best to proceed with this then?

**Comment by StephenGriffies** [comment]
@HiroyukiTsujino  what do you suggest?  Should we back away from making any changes?

**Comment by HiroyukiTsujino** [comment]
@StephenGriffies @durack1 I am offering several possible solutions to the email threads (I have been including both of you to the threads. Let me know if you have not received any.). If we can find a viable solution, I will amend my fork of the repository accordingly and send a pull request. The latest commit to my fork is intended as a practice... apology for the confusion.

**Comment by StephenGriffies** [comment]
Thanks @HiroyukiTsujino    I have been part of the email threads.  Perhaps you should also include @durack1 to keep him in the loop.  We should work through the discussions, which may take some more days.  Sorry @durack1 for the delays...

**Comment by durack1** [comment]
@StephenGriffies @HiroyukiTsujino no problem at all, there is no real urgency with this, unless of course folks want to start their final simulations to contribute. As there is an open dialogue, let's keep the discussion open until a resolution that suits everyone is achieved

**Comment by HiroyukiTsujino** [comment]
@durack1 @StephenGriffies @taylor13 After some discussions over the email threads, I have made the f following proposal, to which I am ending a final call for comments.

**experiment_id = omip1**
```
description = Global ocean - sea-ice coupled experiment forced with the Coordinated
Ocean - ice Reference Experiments inter-annually varying atmospheric and river data sets
for years 1948-2009. Initial ocean tracer fields are based on observations.
Simulation length for at least 5 cycles of the 62-year forcing is required.
The 5-cycle length is recommended to facilitate intercomparison within the experiment 
by using a common simulation length, but a longer simulation length is also accepted.
For each simulation, set the beginning of the simulation (e.g., 1700 and 1638 for 
the 5-cycle and 6-cycle simulation, respectively) as the ‘base time’ of the time axis.
Simulations with different simulation lengths by a single model are treated as members 
of an ensemble. Thus, different 'realization' indexes (e.g., r1, r2, r3, ...) should be used
in a global attribute named 'variant_index' (e.g., r1i1p1f1). It is requested that information
relevant to understanding the differences in members of an ensemble of simulations
is reported in a global attribute named 'variant_info'.
All Priority=1 OMIP diagnostics are requested for all cycles of the 62-year forcing 
to quantify drift. All OMIP diagnostics (Priority=1,2,3) are requested for the last cycle
```
**experiment_id = omip2**

```
min_number_yrs_per_sim = 366 (6-cycles)
```

```
description = Global ocean - sea-ice coupled experiment forced with the JRA55-do 
inter-annually varying atmospheric and river data sets for years 1958-2018.
Initial ocean tracer fields are based on observations. Simulation length for at least 6 cycles
of the 61-year forcing is required.  The 6-cycle length is recommended to facilitate
intercomparison within the experiment by using a common simulation length,
but a longer simulation length is also accepted. In each simulation, set the beginning
of the simulation (e.g., 1653 for the 6-cycle simulation) as the ‘base time’ of the time axis.
Simulations with different simulation lengths by a single model are treated as members
of an ensemble. Thus, different 'realization' indexes (e.g., r1, r2, r3, ...) should be used
in a global attribute named 'variant_index' (e.g., r1i1p1f1). 
It is requested that information relevant to understanding the differences in members of 
an ensemble of simulations is reported in a global attribute named 'variant_info'. 
All Priority=1 OMIP diagnostics are requested for the all cycles of the 61-year forcing
to quantify drift. All OMIP diagnostics (Priority=1,2,3) are requested for the last cycle
```

**experiment_id = omip2-spunup**
```
min_number_yrs_per_sim = 305 (5-cycles)
```
```
description = Same as the omip2 experiment except that it is not initialized with
observed climatologies; rather it is initialized with results from at least a 2000-year
spin up of the coupled physical-biogeochemical models. Another difference from
omip2 is that the required simulation length is 5 cycles of the 61-year forcing.
The spin up simulations may be made with the classic online or offline approach,
or with tracer-acceleration techniques or fast solvers. If an online approach is used,
at the end of the 5th cycle of the JRA55-do forcing, the model's physical fields
should be reinitialized to the values at the start of the 3rd cycle in order to avoid
long-term drift in those fields and to assure that they will not diverge greatly from
physical fields in the omip2 simulation. The spin up also includes radiocarbon
to evaluate deep-ocean circulation
```

**Comment by charliepascoe** [comment]
Thanks @HiroyukiTsujino 
Also, note that information about how ensemble members differ will also be collected by ES-DOC and be made available via the further_info_url.

**Comment by charliepascoe** [comment]
Forgive my ignorance on this question.  For omip2-spunup why would you re-initialize the model's physical fields at the end of the 5th cycle?  Is that not the end of the simulation?

**Comment by charliepascoe** [comment]
> Forgive my ignorance on this question. For omip2-spunup why would you re-initialize the model's physical fields at the end of the 5th cycle? Is that not the end of the simulation?

I get it now, you keep repeating these five cycles throughout the spin-up phase, so of course the reset comes at the end of cycle 5.

**Comment by HiroyukiTsujino** [comment]
Thanks @charliepascoe 
Following sentence has been added to 'description' for  omip1 and 'omip2.

```
This information should also be recorded in the ES-DOC documentation
of each experiment performed by a model and be made available via
the 'further_info_url' attribute. 
```

**Comment by durack1** [comment]
@HiroyukiTsujino thanks for this, I wonder if we should be explicit about the temporal frequency of requested output? As you noted above "All Priority=1 OMIP diagnostics are requested for the all cycles of the 61-year forcing to quantify drift.", to save storage, this could be requested at annual/yearly frequency, rather than monthly which is the default.

The complication with changing the request, is that the data request lists 292 variables for Omon (ocean monthly output), whereas only 122 variables are included in Oyr (ocean annual output).

**Comment by HiroyukiTsujino** [comment]
@durack1 Thanks for reminding us of the data storage issue. The balance between request and storage is also my concern. I talked with several people about the request on Omon variables. Some of the analysts from BGC told me that they are planning to analyse, or at least to take a look at, the priority-1 Omon variables during spinup (i.e., before the last cycle). If the storage allows, I think it would be appropriate to leave the current data request unchanged at this time and to wait until the analysts go through their current analyses and give us feedback about the data request.

**Comment by durack1** [comment]
@HiroyukiTsujino no problem. Each contributing modeling group will need to ascertain an ESGF node (or stand up a node themselves) to publish and host this data, so it is useful to highlight this requirement before large volumes of monthly mean data are generated, but have nowhere to be published (and hosted).

**Comment by HiroyukiTsujino** [comment]
@durack1 @StephenGriffies  How about making the following small correction to 'description'? We simply insert explicit mention about the temporal frequency (Omon, Oyr) for all cycles request.

**experiment_id=omip1**
```
All Priority=1 OMIP diagnostics (Omon, Oyr) are requested for all cycles of the 62-year forcing 
to quantify drift. All OMIP diagnostics (Priority=1,2,3) are requested for the last cycle
```  

**experiment_id=omip2**
```
All Priority=1 OMIP diagnostics (Omon, Oyr) are requested for all cycles of the 61-year forcing
to quantify drift. All OMIP diagnostics (Priority=1,2,3) are requested for the last cycle
```

I think ``Oday`` is not necessary for **all** cycles. I also note that the number of ``Omon`` variables are not too many if we restrict to Priority=1. 

I could amend my commit and issue a pull request, but it would be great if @durack1 could kindly amend the commit in #721 directly.

**Comment by HiroyukiTsujino** [comment]
@durack1  I amended my commit just in case. If a pull request is needed, just let me know.

**Comment by StephenGriffies** [comment]
@HiroyukiTsujino I am fine with your suggested change.

**Comment by durack1** [comment]
@HiroyukiTsujino @StephenGriffies thanks for this, those changes are now in place - please take a peek at [CMIP6_experiment_id.html](https://rawgit.com/WCRP-CMIP/CMIP6_CVs/master/src/CMIP6_experiment_id.html) and let me know of any issues. You *should* be able to sort by `activity id` which will allow all the omip experiments to list sequentially

**Comment by HiroyukiTsujino** [comment]
@durack1 thanks for including the changes. I took a look at CMIP6_experiment_id.html. It seems that the 'description' for omip2 and the entire entry for omip2-spunup are missing from that page. I guess some quotation or parenthesis are missing around 'description' of omip2 because the list ends at omip2.

**Comment by durack1** [comment]
@HiroyukiTsujino it seems the html parsing is having some problems, I have noted that the service we use is deprecated and scheduled for shutdown in October 2019 (#624) and I am not sure what is going wrong.

If you take a peek at the json file, this contains all the (hard to parse by eye) content that you'd expect:

[omip1](https://github.com/WCRP-CMIP/CMIP6_CVs/blob/master/CMIP6_experiment_id.json#L5736-L5762)
[omip1-spunup](https://github.com/WCRP-CMIP/CMIP6_CVs/blob/master/CMIP6_experiment_id.json#L5763-L5788)
[omip2](https://github.com/WCRP-CMIP/CMIP6_CVs/blob/master/CMIP6_experiment_id.json#L5789-L5815)
[omip2-spunup](https://github.com/WCRP-CMIP/CMIP6_CVs/blob/master/CMIP6_experiment_id.json#L5816-L5841)

**Comment by HiroyukiTsujino** [comment]
@durack1 thanks. I took a look at CMIP6_experiment_id.json in the master branch. In the 'description' for omip2,  I found ``as the\u2018base time\u2019of the time axis``. This should read ``as the 'base time' of the time axis``. I guess that there is a problem with the character code, which might be inserted by me. Sorry for this.

**Comment by durack1** [comment]
@HiroyukiTsujino ah ha, you solved my #726 problem! I’ll get onto sorting that next week

**Comment by durack1** [comment]
One further revision is required to `omip2-spunup`:
```
On 7/8/19, 7:03 AM, "Gokhan Danabasoglu" wrote:

    Hiroyuki,
    
    Your understanding is correct. One minor point regarding the revision. Shouldn’t the references to the 3rd and 5th cycles be changed to 4th and 6th cycles? 
    
    Keith: Could you please confirm?
    
    Best,
    Gokhan
    
    > On Jul 8, 2019, at 13:18, Hiroyuki Tsujino wrote:
    > 
    > Dear Gokhan,
    > 
    > Thanks for pointing out this issue. My original proposal on omip2-spunup protocol has actually been included in the CMIP6_CVs about a week ago and now reads as follows:
    > 
    >> "min number of years per simulation" = 305
    >> "description" = 
    >> Same as the omip2 experiment except that it is not initialized
    >> with observed climatologies; rather it is initialized with results
    >> from at least a 2000-year spin up of the coupled physical-biogeochemical
    >> models. Another difference from omip2 is that the required
    >> simulation length is 5 cycles of the 61-year forcing. The spin up
    >> simulations may be made with the classic online or offline
    >> approach, or with tracer-acceleration techniques or fast solvers.
    >> If an online approach is used, at the end of the 5th cycle of the
    >> JRA55-do forcing, the model's physical fields should be reinitialized
    >> to the values at the start of the 3rd cycle in order to avoid
    >> long-term drift in those fields and to assure that they will not 
    >> diverge greatly from physical fields in the omip2 simulation.
    >> The spin up also includes radiocarbon to evaluate deep-ocean circulation
    > 
    > In proposing the above protocol, I was considering that the physical fields will have been also spun up well by the end of an online 2000 year spin-up simulation even if we take the above approach on physical fields (that is, replacing the final physical state of the 5th cycle with the initial physical state of the 3rd cycle before starting another 5-cycle simulation). This may not be necessarily the case, perhaps in an offline approach. Then it would be safer if we require 6-cycle for omip2-spunup. (Gokhan, am I understanding your point correctly?) Also, comparison between omip2 and omip2-spinup would be easier if we require 6-cycle for both.
    > 
    > May I propose the following revision on the protocol for "omip2-spunup"? 
    > Without strong opposition, I would like to ask Paul for including this revision to CMIP6_CVs.
    > 
    >> "min number of years per simulation" = 366
    >> "description" = 
    >> Same as the omip2 experiment except that it is not initialized
    >> with observed climatologies; rather it is initialized with results
    >> from at least a 2000-year spin up of the coupled physical-biogeochemical
    >> models. The spin up simulations may be made with the classic online or
    >> offline approach, or with tracer-acceleration techniques or fast solvers.
    >> If an online approach is used, at the end of the 5th cycle of the
    >> JRA55-do forcing, the model's physical fields should be reinitialized
    >> to the values at the start of the 3rd cycle in order to avoid
    >> long-term drift in those fields and to assure that they will not 
    >> diverge greatly from physical fields in the omip2 simulation.
    >> The spin up also includes radiocarbon to evaluate deep-ocean circulation
    > 
    > Best wishes,
    > 
    > Hiroyuki
    > 
    > 
    > -----Original Message-----
    > From: Gokhan Danabasoglu
    > Sent: Sunday, July 7, 2019 4:31 PM
    > To: Hiroyuki Tsujino
    > 
    > Dear Hiroyuki,
    > 
    > I think I now caught up with all the related emails. I did not see any specific responses to this particular email. Noting that all the subsequent emails are likely not affected since they do not mention omip2-spunup, I would still advocate use of 6 cycles for the omip2-spunup experiments. Although the BGC may be equilibrated, the physical fields are not spun up just as in regular omip2. 
    > 
    > Best,
    > Gokhan
    > 
    >> On Jun 14, 2019, at 10:36, Hiroyuki Tsujino wrote:
    >> 
    >> Hi, 
    >> 
    >> Sorry for bothering you with so many e-mails, but if everyone is fine with the proposal on the new protocols, I will issue a request on next Monday that the revised protocols are reflected in the following page:
    >> <https://rawgit.com/WCRP-CMIP/CMIP6_CVs/master/src/CMIP6_experiment_id.html>
    >> 
    >> Note that I set 5-cycles for omip2-spunup considering that the fields have been already spun up.
    >> Silence would be considered as consent.
    >> 
    >> As soon as the revised protocol is reflected in the above page, I will send a notice to the cmip6-omip mailing list.
    >> 
    >> Best wishes,
    >> 
    >> Hiroyuki 
```
